spec/dynamoid/persistence_spec.rb

```diff
@@ -2597,6 +2597,38 @@ def log_message
         expect { a1.destroy }.to_not raise_error
       end
     end
+
+    context 'with associations' do
+      let!(:user) { User.create }
+      let!(:camel_case) { user.camel_case.create }
+      let!(:magazine) { user.books.create }
+      let!(:monthly) { user.monthly.create }
+      let!(:subscription) { user.subscriptions.create }
+
+      it 'updates belongs_to association' do
+        expect do
+          user.delete
+        end.to change { CamelCase.find(camel_case.id).users.target }.from([user]).to([])
+      end
+
+      it 'updates has_many association' do
+        expect do
+          user.delete
+        end.to change { Magazine.find(magazine.title).owner.target }.from(user).to(nil)
+      end
+
+      it 'updates has_one association' do
+        expect do
+          user.delete
+        end.to change { Subscription.find(monthly.id).customer.target }.from(user).to(nil)
+      end
```

Comment:
The only way this fix could be improved is to ignore `RecordNotFound` errors when association is loaded and there are some broken foreign keys (what is described in the original issue https://github.com/Dynamoid/dynamoid/issues/467) . This way we would solved the problem for existing corrupted references.

But it's out of scope of this PR and it's up to you.